org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/builder/BatchImageBuilder.java

```diff
@@ -348,4 +363,50 @@ protected void storeProblemsFor(SourceFile sourceFile, CategorizedProblem[] prob
 public String toString() {
 	return "batch image builder for:\n\tnew state: " + this.newState; //$NON-NLS-1$
 }
+
+@Override
+public void startBatch() {
+	this.batchMode = true;
+}
+
+@Override
+protected void writeClassFileContents(ClassFile classFile, IFile file, String qualifiedFileName, boolean isTopLevelType,
+		SourceFile compilationUnit) throws CoreException {
+	if (JavaBuilder.DEBUG) {
+		trace("Writing changed class file " + file.getName());//$NON-NLS-1$
+	}
+	byte[] content = classFile.getBytes();
+	if (this.batchMode) {
+		// flush before 100MB to avoid OOME:
+		if (!this.classContents.isEmpty() && this.classContentsBytesQueued + content.length >= 100_000_000) {
```

Comment:
Could this please be a constant? CLASS_CONTENT_CACHE_LIMIT or something?

Reply:
done - MAX_CLASS_CONTENTS_BYTES_QUEUED